### PR TITLE
Add type hints into optuna/cli.py

### PR DESCRIPTION
**Pull request by airyou**
## Motivation

Please refer to the Issue #711 .

## Description of the changes

I added type hints into optuna/cli.py

### CODE REVIEW

File: optuna/cli.py [hunk 1]
```diff
@@ -20,30 +20,26 @@
 from cliff.commandmanager import CommandManager
 from cliff.lister import Lister
 
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
```
Comment (hvy) [inline optuna/cli.py:27]: Could you place these imports from `typing` with the other standard library imports at the top (after `import types`)?

**Review by hvy — CHANGES_REQUESTED** [review]
Left a minor comment but changes otherwise look great. Thanks for the PR!

**Review by toshihikoyanase — COMMENTED** [review]
Thank you for your contribution. LGTM except for the comment https://github.com/optuna/optuna/pull/1648#pullrequestreview-469932340.

File: optuna/cli.py (hunk 1 above)
Comment (airyou) [reply optuna/cli.py:27]: Thank you for your review. I fixed code.

**Review by hvy — APPROVED** [review]
LGTM!

**Review by toshihikoyanase — APPROVED** [review]
LGTM!

**Pull request merged**